core/backend-itwin-client/src/test/AgentAuthorizationClient.test.ts

```diff
@@ -64,49 +63,47 @@ describe("AgentAuthorizationClient (#integration)", () => {
     chai.expect(issuer.introspection_endpoint).equals(`${url}/connect/introspect`);
   });
 
-  it("should get valid OIDC tokens for agent applications", async () => {
-    const agentClient = new AgentAuthorizationClient(agentConfiguration);
-    const now = Date.now();
-    const jwt: AccessToken = await agentClient.getAccessToken(requestContext);
-
-    const expiresAt = jwt.getExpiresAt();
-    chai.assert.isDefined(expiresAt);
-    chai.assert.isAbove(expiresAt!.getTime(), now);
-
-    const startsAt = jwt.getStartsAt();
-    chai.assert.isDefined(startsAt);
-    chai.assert.isAtLeast(startsAt!.getTime(), expiresAt!.getTime() - 1 * 60 * 60 * 1000); // Starts atleast 1 hour before expiry
-
-    await validator.validateContextRegistryAccess(jwt);
-    await validator.validateIModelHubAccess(jwt);
-
-    const refreshJwt: AccessToken = await agentClient.getAccessToken(requestContext);
-    await validator.validateContextRegistryAccess(refreshJwt);
-    await validator.validateIModelHubAccess(refreshJwt);
-  });
-
-  it("should not refresh token unless necessary", async () => {
-    const agentClient = new AgentAuthorizationClient(agentConfiguration);
-
-    const jwt: AccessToken = await agentClient.getAccessToken(requestContext);
-
-    // Refresh after a second, and the token should remain the same
-    await BeDuration.wait(1000);
-    let refreshJwt: AccessToken = await agentClient.getAccessToken(requestContext);
-    chai.assert.strictEqual(refreshJwt, jwt);
-
-    // Set the expiry of the token to be 2 min from now, and the token should remain the same
-    const twoMinFromNow = new Date(Date.now() + 2 * 60 * 1000);
-    const jwtExpiresAtTwoMinFromNow = new AccessToken(jwt.toTokenString(IncludePrefix.No), jwt.getStartsAt(), twoMinFromNow, jwt.getUserInfo());
-    (agentClient as any)._accessToken = jwtExpiresAtTwoMinFromNow;
-    refreshJwt = await agentClient.getAccessToken(requestContext);
-    chai.assert.strictEqual(refreshJwt, jwtExpiresAtTwoMinFromNow);
-
-    // Set the expiry of the token to be less than a min from now, and the token should be refreshed
-    const lessThanMinFromNow = new Date(Date.now() + 59 * 1000);
-    const jwtExpiresAtLessThanMinFromNow = new AccessToken(jwt.toTokenString(IncludePrefix.No), jwt.getStartsAt(), lessThanMinFromNow, jwt.getUserInfo());
-    (agentClient as any)._accessToken = jwtExpiresAtLessThanMinFromNow;
-    refreshJwt = await agentClient.getAccessToken(requestContext);
-    chai.assert.notStrictEqual(refreshJwt.toTokenString(IncludePrefix.No), jwtExpiresAtLessThanMinFromNow.toTokenString(IncludePrefix.No));
-  });
+  // it("should get valid OIDC tokens for agent applications", async () => {

```

Comment:
What were some of the issues with these tests? Ideally we'd get them working prior to merging this PR.

Reply:
I guess kind of in relation to my comment/question on `PushChanges.test.ts`, how should we be mocking the expiration date for this?